my-app/src/ListIterator.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class ListIterator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      listItems: this.props.listItems
+    };
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (prevState.listItems !== nextProps.listItems) {
+      let sortedArray = nextProps.listItems.sort();
+      return {
+        listItems: sortedArray
+      };
+    }
+    return null;
+  }
+
+
+  handleDelete = event => {
+    let clickedButtonElement = event.currentTarget.previousElementSibling.textContent;
```

Comment:
You can directly pass the value or index to bind function instead of getting it from sibling element.